### PR TITLE
SIMSBIOHUB-356: Update survey list endpoint to return a subset of survey fields

### PR DESCRIPTION
**Pull request by NickPhura**
## Links to Jira Tickets

https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-356

#### Replace existing survey/list endpoint with survey/index.
- Add service/repo function to fetch a smaller subset of survey fields for a given project id.
- Call new service function

#### Update useSurveyApi
- Calls survey/index endpoint
- Update related interface to match new endpoint response
#### Update survey list component
- No longer renders a column for survey purpose. There is some work being done to modify this data (remove fields, make others arrays, etc, so for now I just opted to just ignore it entirely until it stabilizes)

## Testing Notes

Loading the project page (specifically the list of surveys) works. It should also load much faster now, since it isn't loading every survey field or the survey spatial (which could cause major slowdowns when loading multiple surveys).

### CODE REVIEW

File: api/src/paths/project/{projectId}/survey/index.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    {
+      Bearer: []
+    }
+  ],
+  parameters: [
+    {
+      in: 'path',
+      name: 'projectId',
+      schema: {
+        type: 'number'
```
Comment (GrahamS-Quartech) [inline api/src/paths/project/{projectId}/survey/index.ts]: nit: integer

File: api/src/paths/project/{projectId}/survey/index.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            items: {
+              type: 'object',
+              required: ['surveyData', 'surveySupplementaryData'],
+              properties: {
+                surveyData: {
+                  type: 'object',
+                  required: ['survey_id', 'name', 'start_date', 'end_date', 'focal_species', 'focal_species_names'],
+                  properties: {
+                    survey_id: {
+                      type: 'number',
```
Comment (GrahamS-Quartech) [inline api/src/paths/project/{projectId}/survey/index.ts]: nit: integer

File: api/src/repositories/survey-repository.ts [hunk 3]
```diff
@@ -119,6 +119,17 @@ export type StakeholderPartnershipRecord = z.infer<typeof StakeholderPartnership
 
 export type IndigenousPartnershipRecord = z.infer<typeof IndigenousPartnershipRecord>;
 
+export const SurveyBasicFields = z.object({
```
Comment (curtisupshall) [inline api/src/repositories/survey-repository.ts:122]: +1 for more verbose naming for survey fields. 👍

**Review by curtisupshall — COMMENTED** [review]
Code looks good! Just needs tests fixed

**Comment by GrahamS-Quartech** [comment]
Tested in the deployed pod and added a bunch of surveys. Very snappy. 👍

File: api/src/paths/project/{projectId}/survey/index.ts (hunk 1 above)
Comment (NickPhura) [reply api/src/paths/project/{projectId}/survey/index.ts]: Fixed

File: api/src/paths/project/{projectId}/survey/index.ts (hunk 2 above)
Comment (NickPhura) [reply api/src/paths/project/{projectId}/survey/index.ts]: Fixed

**Review by curtisupshall — APPROVED** [review]
tested, looks good to me!

**Review by al-rosenthal — APPROVED** [review]
💎

**Pull request merged**